File: example/java/config/ai/Ai_condition.java
package config.ai;

public class Ai_condition {
    private int iD;
    private String desc;
    private int formulaID;
    private java.util.List<Integer> argIList;
    private java.util.List<Integer> argSList;

    private Ai_condition() {
    }

    public static Ai_condition _create(configgen.genjava.ConfigInput input) {
        Ai_condition self = new Ai_condition();
        self.iD = input.readInt();
        self.desc = input.readStr();
        self.formulaID = input.readInt();
        self.argIList = new java.util.ArrayList<>();
        for (int c = input.readInt(); c > 0; c--) {
            self.argIList.add(input.readInt());
        }
        self.argSList = new java.util.ArrayList<>();
        for (int c = input.readInt(); c > 0; c--) {
            self.argSList.add(input.readInt());
        }
        return self;
    }

    /**
     * ID
     */
    public int getID() {
        return iD;
    }

    /**
     * 描述
     */
    public String getDesc() {
        return desc;
    }

    /**
     * 公式
     */
    public int getFormulaID() {
        return formulaID;
    }

    /**
     * 参数(int)1
     */
    public java.util.List<Integer> getArgIList() {
        return argIList;
    }

    /**
     * 参数(string)1
     */
    public java.util.List<Integer> getArgSList() {
        return argSList;
    }

    @Override
    public String toString() {
        return "(" + iD + "," + desc + "," + formulaID + "," + argIList + "," + argSList + ")";
    }

    public static Ai_condition get(int iD) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.ai_ai_condition_All.get(iD);
    }

    public static java.util.Collection<Ai_condition> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.ai_ai_condition_All.values();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Ai_condition self = Ai_condition._create(input);
                mgr.ai_ai_condition_All.put(self.iD, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            // no resolve
        }

    }

}

File: example/go/config/Ai/Ai_condition.go
package ai

type Ai_condition struct {
    iD int
    desc string //描述
    formulaID int //公式
    argIList []int //参数(int)1
    argSList []int //参数(string)1
}

//getters
func (t *Ai_condition) GetID() int {
    return t.iD
}

func (t *Ai_condition) GetDesc() string {
    return t.desc
}

func (t *Ai_condition) GetFormulaID() int {
    return t.formulaID
}

func (t *Ai_condition) GetArgIList() []int {
    return t.argIList
}

func (t *Ai_condition) GetArgSList() []int {
    return t.argSList
}

var all []Ai_condition
func GetAll() []Ai_condition {
    return all[:len(all)]:len(all)]
}

var allMap map[int]Ai_condition
func Get(key int) (Ai_condition,bool){
    return allMap[key]
}
        public static Ai_condition Get(int iD)
        {
            Ai_condition v;
            return all.TryGetValue(iD, out v) ? v : null;
        }

        public static List<Ai_condition> All()
        {
            return all.OrderedValues;
        }

        public static List<Ai_condition> Filter(Predicate<Ai_condition> predicate)
        {
            var r = new List<Ai_condition>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<int, Ai_condition>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.ID, self);
            }
        }

